docs/rfc-1-staking-contract.adoc

```diff
@@ -0,0 +1,381 @@
+future checks of the stake amount.
+
+Top-ups can be executed for native T stakes and for legacy KEEP/NU stakes.
+
+Native T stakers can only top-up their stakes with a liquid T.
+
+Existing KEEP and NU stakers wanting to execute a top-up have two options. One
+option is to wrap their KEEP/NU to T and then, execute a top-up in T staking
+contract. The second option is to execute a top-up in their legacy staking
+contracts and notify T staking contract about the fact their legacy stake
+increased.
+
+Effectively, it means that existing KEEP stakers can mix their legacy KEEP
+stakes with liquid T stakes. Similarly, existing NU stakers can mix their legacy
+NU stakes with liquid T stakes. This functionality adds some complexity to the
+staking contract but it puts existing KEEP/NU stakers in the same spot as new T
+stakers. Without it, existing stakers would not be able to top-up their stakes
+with T earned from operating in the network, so they would be in a worse spot
+than new T stakers allowed to top-up their stakes using earned T and this way
+increasing their rewards.
+
+Anyone can execute a stake top-up for an operator using a liquid T. Stake top-up
+does not automatically increase authorization levels for applications.
+
+==== Undelegating a stake
+
+The owner or operator may choose to undelegate the stake by submitting an
+undeleagation order. Stake delegation should be marked in the staking contract
+as undelegating. Always the entire stake is getting undelegated and there is a
+fixed, non-governable, two months undelegation period. All applications should
+be designed in such a way that a two-month undelegation period is sufficient to
+retire all groups undelegating operators are involved in. The operator is not
+eligible for any new work during the stake undelegation period but it is still
+required to finish all the pending work. Authorized applications are eligible
+to slash the operator for the entire undelegation period.
+
+==== Recovering a stake
+
+Once the undelegation period finishes, anyone - even a third party - can recover
+the stake. Staking contract sets the stake as recovered and returns all
+delegated tokens to the owner. Applications shouldn't be able to slash after
+undelegation is complete, even before the stake is returned. Information about
+owner, operator, and beneficiary, is retained in the contract in case some
+application rewards are still available for withdrawal.
+
+==== Keeping information in sync
+
+To avoid expensive calls to legacy staking contracts, it is assumed that
+information in T staking contract about the staked amount is always up-to-date.
+By design, this is always the case for native T stakers but might not be the
+case for legacy KEEP/NU stakers.
+
+T staking contract should expose a function allowing to seize some amount of T
+from the operator whose stake in T contract is lower than their stake in the
+legacy staking contract. 5% of the amount seized is given to the person who
```

Comment:
Because we can slash only up to allocated amount (per app) then if legacy contract has more stake than allocated it's not malicious behaviour but part of allocating process. Maybe I just misunderstood this sentence

Reply:
It should be `whose stake in the T contract is higher than their stake in the legacy staking contract`; the intent is to cover the situation where someone gets slashed on legacy side but doesn't update their now lower stake to the T contract and thus benefits.

Reply:
@eth-r is right, this is a typo: s/lower/higher